// ngp-sort-pipe/__ivy_ngcc__/fesm2015/ngp-sort-pipe.js
import { __decorate } from 'tslib';
import { Pipe, NgModule } from '@angular/core';
import { orderBy } from 'lodash';

import * as ɵngcc0 from '@angular/core';
let SortByPipe = class SortByPipe {
    transform(value, order = '', column = '') {
        if (!value || order === '' || !order) {
            return value;
        } // no array
        if (!column || column === '') {
            if (order === 'asc') {
                return value.sort();
            }
            else {
                return value.sort().reverse();
            }
        } // sort 1d array
        if (value.length <= 1) {
            return value;
        } // array with only one item
        return orderBy(value, [column], [order]);
    }
};
SortByPipe.ɵfac = function SortByPipe_Factory(t) { return new (t || SortByPipe)(); };
SortByPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "sortBy", type: SortByPipe, pure: true });

let NgpSortModule = class NgpSortModule {
};
NgpSortModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgpSortModule });
NgpSortModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgpSortModule_Factory(t) { return new (t || NgpSortModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SortByPipe, [{
        type: Pipe,
        args: [{ name: 'sortBy' }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgpSortModule, { declarations: [SortByPipe], exports: [SortByPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgpSortModule, [{
        type: NgModule,
        args: [{
                declarations: [SortByPipe],
                exports: [SortByPipe]
            }]
    }], null, null); })();

/*
 * Public API Surface of ngp-sort
 */

/**
 * Generated bundle index. Do not edit.
 */

export { NgpSortModule, SortByPipe };
